### DISCUSSION TITLE
How to terminate capture?

### DISCUSSION THREAD

**Discussion by derekchen60** [streamlink/streamlink#4616] [Q&A] [answered]
Hi all,

I'm trying to figure out if there is a way to terminate the capture without using ctrl + c. Whenever I use ctrl + c, it often corrupts the video and renders huge chunks of it unusable. Is there a way the capture is supposed to be terminated?

Thanks!

**Comment by bastimeyer** [comment, marked as answer]
HLS streams can be terminated after a certain duration:
https://streamlink.github.io/cli.html#cmdoption-hls-duration

Sending SIGINT via CTRL+C shouldn't cause any corruption to the written stream though. And in any case, you can simply remux the data via FFmpeg.

**Comment by derekchen60** [comment]
Thanks for the reply. It's interesting because I noticed that if I keep the stream video file open, then often the program terminates properly when the stream ends. If I don't keep the file open, I end up with a lot of corruption to the written file. I guess it's just my workaround.
